diff --git a/win32/Makefile.msvc b/win32/Makefile.msvc
index 54e8e23..c7cab76 100644
--- a/win32/Makefile.msvc
+++ b/win32/Makefile.msvc
@@ -59,7 +59,7 @@ CFLAGS = $(CFLAGS) /D_CRT_SECURE_NO_DEPRECATE /D_CRT_NONSTDC_NO_DEPRECATE
 LD = link.exe
 LDFLAGS = /nologo
 LDFLAGS = $(LDFLAGS) /LIBPATH:$(BINDIR) /LIBPATH:$(LIBPREFIX)
-LIBS = wsock32.lib
+LIBS = 
 
 # The archiver and its options.
 AR = lib.exe
@@ -225,7 +225,7 @@ $(XSLT_INTDIR_A) :
 
 # An implicit rule for static libxslt compilation.
 {$(XSLT_SRCDIR)}.c{$(XSLT_INTDIR_A)}.obj::
-	$(CC) $(CFLAGS) /D "LIBXML_STATIC" /D "LIBXSLT_STATIC" \
+	$(CC) $(CFLAGS) /D "LIBXSLT_STATIC" \
 		/Fo$(XSLT_INTDIR_A)\ /c $<
 
 # Compiles libxslt source. Uses the implicit rule for commands.
@@ -271,7 +271,7 @@ $(EXSLT_INTDIR_A) :
 
 # An implicit rule for static libexslt compilation.
 {$(EXSLT_SRCDIR)}.c{$(EXSLT_INTDIR_A)}.obj::
-	$(CC) /D "LIBXML_STATIC" /D "LIBXSLT_STATIC" /D "LIBEXSLT_STATIC" \
+	$(CC) /D "LIBXSLT_STATIC" /D "LIBEXSLT_STATIC" \
 		/I$(EXSLT_SRCDIR) $(CFLAGS) /Fo$(EXSLT_INTDIR_A)\ /c $<
 
 # Compiles libxslt source. Uses the implicit rule for commands.
@@ -310,7 +310,7 @@ $(UTILS_INTDIR) :
 # An implicit rule for xsltproc and friends.
 APPLIBS = $(LIBS)
 !if "$(STATIC)" == "1"
-APPLIBS = $(LIBS) libxml2_a.lib
+APPLIBS = $(LIBS) libxml2.lib
 !if "$(WITH_ICONV)" == "1"
 APPLIBS = $(APPLIBS) iconv.lib
 !endif
@@ -318,7 +318,7 @@ APPLIBS = $(APPLIBS) iconv.lib
 APPLIBS = $(APPLIBS) zlib.lib
 !endif
 {$(UTILS_SRCDIR)}.c{$(BINDIR)}.exe:
-	$(CC) /D "LIBXML_STATIC" /D "LIBXSLT_STATIC" /D "LIBEXSLT_STATIC" \
+	$(CC) /D "LIBXSLT_STATIC" /D "LIBEXSLT_STATIC" \
 		$(CFLAGS) /Fo$(UTILS_INTDIR)\ /c $< 
 	$(LD) $(LDFLAGS) /OUT:$@ $(XSLT_A) $(EXSLT_A) $(APPLIBS) $(UTILS_INTDIR)\$(<B).obj
 	@$(_VC_MANIFEST_EMBED_EXE)
